srv/components/misc_pkgs/mocks/cortx/files/scripts/buildrpm.sh

```diff
@@ -1,26 +1,62 @@
 #!/bin/bash
 
-VERSION="2.0.0"
-
-WORKING_DIR=${PWD}/../
-BUILD_DIR=$(realpath ~)
-
-cd ${BUILD_DIR}
-
-rm -rf ${BUILD_DIR}/rpmbuild
-
-rpmdev-setuptree
-
-cp -r ${WORKING_DIR}/specs/* "${BUILD_DIR}/rpmbuild/SPECS/"
-
-cd "$BUILD_DIR/rpmbuild/"
-ls SPECS/ | sed -E "s/(.*).spec/\1-${VERSION}/" | xargs -I{} mkdir -p ./SOURCES/{}
-ls SOURCES/ | xargs -I{} touch SOURCES/{}/mock.txt
-
-cd SOURCES/
-ls ./ | xargs -I{} tar czf {}.tar.gz {}
-
-cd ../
-ls SPECS/ | xargs -I{} rpmbuild -ba SPECS/{}
-
-cd ${WORKING_DIR}
+set -eu
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+VERSION="${1:-2.0.0}"
+BUILD_DIR="${2:-(realpath ~)}"
+SPEC=cortx-mock.spec
+SPEC_NO_API=cortx-mock-no-api.spec
+
+WORKING_DIR="$(realpath $script_dir/..)"
+
+# TODO a kind of duplication for pillar/components/commons.sls
+# TODO provisioner pkgs
+PKG_COMP_LIST=(
+    "cortx-csm_agent,csm"
+    "cortx-hare,hare"
+    "cortx-motr,motr"
+    "cortx-s3server,s3"
+    "cortx-ha,ha"
+    "cortx-sspl,sspl"
+    "uds-pyi,uds"
+    "cortx-cli,"
+    "cortx-sspl-test,"
+    "cortx-s3iamcli,"
+    "cortx-csm_web,"
+)
+
+rm -rf "${BUILD_DIR}"/rpmbuild
+mkdir -p "${BUILD_DIR}"/rpmbuild/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+
+pushd ${BUILD_DIR}
+    cp ${WORKING_DIR}/specs/cortx-mock.spec "rpmbuild/SPECS/${SPEC}"
+    grep -v setup.yaml ${WORKING_DIR}/specs/cortx-mock.spec >"rpmbuild/SPECS/${SPEC_NO_API}"
```

Comment:
I think it is better to add a one-line comment to explain in which cases we need `cortx-mock-no-api.spec` without installing `setup.yaml`. For me, it is not obvious when the variable "$comp" will be empty.

Reply:
done